File: dotnet/Multiplayer/Rooms/RoomJoinRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Rooms
{
    /// <summary>
    /// Request data for joining a game room.
    /// </summary>
    /// <typeparam name="T">The type of optional player data to include.</typeparam>
    public class RoomJoinRequest<T> where T : class, new()
    {
        [JsonInclude]
        private string? Password { get; set; }
        [JsonInclude]
        private T? Player_data { get; set; }

        /// <summary>
        /// Initializes a new RoomJoinRequest.
        /// </summary>
        /// <param name="password">Optional password for the room.</param>
        /// <param name="playerData">Optional player data to include.</param>
        public RoomJoinRequest(string? password = null, T? playerData = null)
        {
            this.Password = password;
            this.Player_data = playerData;
        }
    }
}

File: java/src/main/java/com/michitai/multiplayer/rooms/RoomJoinRequest.java
package com.michitai.multiplayer.rooms;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Request for joining an existing game room.
 */
public class RoomJoinRequest {
    @JsonProperty("password")
    private String password;

    @JsonProperty("player_data")
    private String playerData;

    public RoomJoinRequest(String password, String playerData) {
        this.password = password;
        this.playerData = playerData;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getPlayerData() {
        return playerData;
    }

    public void setPlayerData(String playerData) {
        this.playerData = playerData;
    }
}
